shell/components/AppModal.vue

```diff
@@ -56,6 +57,20 @@ export default defineComponent({
     name: {
       type:    String,
       default: '',
+    },
+    /**
+     * Modal visibility (used for focus-trap)
+     */
+    modalVisibility: {
+      type:    Boolean,
+      default: false,
+    },
```

Comment:
The `modalVisibility` looks redundant to me.. Shouldn't modal instances be conditionally rendered already?

Reply:
It's not about the actual render of the modal itself, but we need to propagate a variable related to the visibility of the modal so it can act as a "trigger" (in this case a watcher) for the focus trap, as explained in https://github.com/rancher/dashboard/pull/13176#discussion_r1945097081